commit 24a74fe601243ab146dda6f3bc466f34db830225
Author: Peter Maydell <peter.maydell@linaro.org>
Date:   Fri Aug 9 17:07:19 2013 +0100

    target-arm: Allow raw_read() and raw_write() to handle 64 bit regs
    
    Extend the raw_read() and raw_write() helper accessors so that
    they can be used for 64 bit registers as well as 32 bit registers.
    
    Signed-off-by: Peter Maydell <peter.maydell@linaro.org>
    Tested-by: Laurent Desnogues <laurent.desnogues@gmail.com>

Index: qemu/target-arm/helper.c
===================================================================
--- qemu.orig/target-arm/helper.c	2013-10-09 15:38:11.238456453 -0500
+++ qemu/target-arm/helper.c	2013-10-09 15:38:11.230456452 -0500
@@ -67,14 +67,22 @@
 static int raw_read(CPUARMState *env, const ARMCPRegInfo *ri,
                     uint64_t *value)
 {
-    *value = CPREG_FIELD32(env, ri);
+    if (ri->type & ARM_CP_64BIT) {
+        *value = CPREG_FIELD64(env, ri);
+    } else {
+        *value = CPREG_FIELD32(env, ri);
+    }
     return 0;
 }
 
 static int raw_write(CPUARMState *env, const ARMCPRegInfo *ri,
                      uint64_t value)
 {
-    CPREG_FIELD32(env, ri) = value;
+    if (ri->type & ARM_CP_64BIT) {
+        CPREG_FIELD64(env, ri) = value;
+    } else {
+        CPREG_FIELD32(env, ri) = value;
+    }
     return 0;
 }
 
